src/index.ts

```diff
@@ -181,22 +273,35 @@ export class Authenticator {
     const decoded = await this._jwtVerifier.verify(tokens.idToken);
     const username = decoded['cognito:username'] as string;
     const usernameBase = `${this._cookieBase}.${username}`;
+    const cookieDomain = getCookieDomain(domain, this._disableCookieDomain, this._cookieDomain);
     const cookieAttributes: CookieAttributes = {
-      domain: this._disableCookieDomain ? undefined : domain,
+      domain: cookieDomain,
       expires: new Date(Date.now() + this._cookieExpirationDays * 864e+5),
       secure: true,
       httpOnly: this._httpOnly,
       sameSite: this._sameSite,
       path: this._cookiePath,
     };
     const cookies = [
-      Cookies.serialize(`${usernameBase}.accessToken`, tokens.accessToken, cookieAttributes),
-      Cookies.serialize(`${usernameBase}.idToken`, tokens.idToken, cookieAttributes),
-      ...(tokens.refreshToken ? [Cookies.serialize(`${usernameBase}.refreshToken`, tokens.refreshToken, cookieAttributes)] : []),
+      Cookies.serialize(`${usernameBase}.accessToken`, tokens.accessToken, this._getOverridenCookieAttributes(cookieAttributes, 'accessToken')),
+      Cookies.serialize(`${usernameBase}.idToken`, tokens.idToken, this._getOverridenCookieAttributes(cookieAttributes, 'idToken')),
+      ...(tokens.refreshToken ? [Cookies.serialize(`${usernameBase}.refreshToken`, tokens.refreshToken, this._getOverridenCookieAttributes(cookieAttributes, 'refreshToken'))] : []),
       Cookies.serialize(`${usernameBase}.tokenScopesString`, 'phone email profile openid aws.cognito.signin.user.admin', cookieAttributes),
       Cookies.serialize(`${this._cookieBase}.LastAuthUser`, username, cookieAttributes),
     ];
 
+    // Clear CSRF Token Cookies
+    if (this._csrfProtection) {
```

Comment:
I believe, location decoding and validation is missing here

Reply:
Could you elaborate on what you mean by location decoding/validation?

The purpose of `_getRedirectResponse` is to set auth token cookies (id, access and refresh) after they have been fetched from Cognito. In addition, when CSRF protection is enabled, it'd also clear the CSRF cookies (nonce, nonce hmac and pkce). These cookies were initially set during the authorize step `_getRedirectToCognitoUserPoolResponse`

Reply:
I mean if CSRF protection is enabled, user is redirected to encoded path (e.g. `/eyJub25...`) after successful login. So `requestParams.state` needs to be decoded first before calling `_getRedirectResponse()` inside `handle()`

Reply:
Done